### PR TITLE
add onERC721Received support

### PR DESCRIPTION
**Pull request by kamescg**
https://github.com/code-423n4/2021-06-pooltogether-findings/issues/118

### CODE REVIEW

File: contracts/prize-pool/PrizePool.sol [hunk 1]
```diff
@@ -1050,6 +1051,15 @@ abstract contract PrizePool is PrizePoolInterface, OwnableUpgradeable, Reentranc
...
     }
   }
+  
+  /// @notice Required for ERC721 safe token transfers from smart contracts.
+  /// @param operator The address that acts on behalf of the owner
+  /// @param from The current owner of the NFT
+  /// @param tokenId The NFT to transfer
+  /// @param data Additional data with no specified format, sent in call to `_to`.
+  function onERC721Received(address operator, address from, uint256 tokenId, bytes calldata data) external override returns (bytes4){
+    return this.onERC721Received.selector;
```
Comment (PierrickGT) [inline contracts/prize-pool/PrizePool.sol:1061]: They recommend to use `IERC721.onERC721Received.selector` in the doc but I suppose it also works with `this`, maybe a bit less explicit though.
https://docs.openzeppelin.com/contracts/3.x/api/token/erc721#IERC721Receiver-onERC721Received-address-address-uint256-bytes-

**Review by PierrickGT — APPROVED** [review]

**Review by asselstine — CHANGES_REQUESTED** [review]
It'd be nice to see a test for this

File: contracts/prize-pool/PrizePool.sol (hunk 1 above)
Comment (kamescg) [reply contracts/prize-pool/PrizePool.sol:1061]: Updated to `IERC721ReceiverUpgradeable.onERC721Received.selector`

**Comment by kamescg** [comment]
@asselstine The onERC721Received test has been added to the PrizePool.test.js file - checking balanceOf of update and event emission.

**Review by asselstine — APPROVED** [review]

**Pull request merged**